src/server/server_family.cc

```diff
@@ -448,6 +449,15 @@ void ServerFamily::Shutdown() {
     snapshot_fiber_.Join();
   }
 
+  if (absl::GetFlag(FLAGS_save_on_shutdown)) {
+    pb_task_->Await([this] {
```

Comment:
while technically you can use `pb_task_` it's a bit confusing because we keep it because we need to know on which proactor we run periodic tasks.
You can use `shard_set->pool()->GetNextProactor();` to pick a proactor to run whatever you need.